studio/app/common/core/storage/s3_storage_controller.py

```diff
@@ -51,114 +46,50 @@ def make_experiment_remote_path(self, workspace_id: str, unique_id: str) -> str:
 
     def download_all_experiments_metas(self) -> bool:
         """
-        NOTE:
-          - この処理（config yaml files の S3 からのダウンロード）では、
-            ダウンロード対象ファイルリストの取得に、python module (boto3) ではなく、
-            外部コマンド (aws cli) を利用している
-          - aws cli を利用する事由
-            1. boto3 では、取得対象のファイルリストの Server(AWS) Side でのfilterをサポートしていない（2024.7時点）
-                - 「Prefix配下のファイルリストをすべて取得 → Client Side でのFilter」の操作手順となる
-            2. また 1. の操作を行う場合、Pagination の考慮も必要となる
-          - 上記のため、ファイルリストの取得には、aws cli を利用する形式としている
-            - aws cli (`aws s3 ls`) も、基本的には Server(AWS) Side のファイルリストfilterには非対応だが、
-              `aws s3 sync` の利用により、間接的に Server(AWS) Side での filterが利用可能
-            - なお aws cli の利用は暫定的な対応であり、最終的には boto3 でfilterを実現できることが望ましい
+        Download all experiment metadata files from S3 using boto3.

```

Comment:
The reason for using `aws s3 sync` is, as stated in the comment, “to have the filter performed on the Server (AWS) Side”.

Since optinist may generate a large number of files, I care about avoiding useless filters on the client side.

From the above, it is better to estimate the maximum number of files that can actually be generated and compare the performance between the `aws s3 sync` version and the boto3 version.

Reply:
どちらの方法がいいかclaude.aiに聞いたんですが、この通りで答えました。boto3を使用した改訂版がAWS CLIのs3 syncよりも優れている理由は以下の通りです：

1. 依存関係の削減：boto3はPythonのAWS SDKであり、追加のCLIツールをインストールする必要がありません。これにより、システムの依存関係が減少し、環境の管理が容易になります。
2. セキュリティの向上：boto3はプログラム内で直接AWS認証情報を管理できるため、シェルコマンドで認証情報を扱うよりも安全です。
3. パフォーマンスの最適化：boto3を使用すると、必要なファイルのみを効率的にフィルタリングし、ダウンロードすることができます。これにより、不要なネットワークトラフィックを削減できる可能性があります。
4. エラーハンドリングの改善：boto3を使用すると、Pythonのtry-except文を用いて、きめ細かいエラーハンドリングが可能になります。これにより、問題が発生した際の対応がより柔軟になります。
5. プログラマティックな制御：boto3を使用すると、S3操作をよりきめ細かく制御でき、アプリケーションのニーズに合わせてカスタマイズしやすくなります。
これらの理由により、boto3を使用した改訂版は、多くの場合においてAWS CLIのs3 syncコマンドよりも優れた選択肢となり得ます。

もちろん、川田さんの判断にお任せしています

Reply:
はい、基本的には外部コマンドに依存せず、pythonコードで完結できる形が望ましいと思います。
しかし、boto3（S3 API）が、`aws s3 sync` に相当する機能を提供していないため、まずは `aws s3 sync` を利用した経緯です。

先のメッセージで記載したように、性能比較も行い、方式をFixできると良いと考えます。

※claudeの回答のうち、# 3 などは、aws cliでも同等そうですね。